packages/salesforcedx-sobjects-faux-generator/src/generator/fauxClassGenerator.ts

```diff
@@ -189,7 +189,13 @@ export class FauxClassGenerator {
         err.stack
       );
     }
-    const filteredSObjects = sobjects.filter(this.isRequiredSObject);
+
+    // when run manually does not exclude any object
+    const filteredSObjects =
```

Comment:
Since we have 3 different options when running `SFDX: Refresh SObject Definitions` (All SObjects, Custom SObjects & Standard SObjects), I think we should update this to be exclusive to `All SObjects` and for everything else to still use the filter.

To provide some context, we introduced the use of an sobject filter to reduce the amount of synced objects since an org without any customizations can have between 400-600 sobjects. This also helps the Apex language server since the faux classes generated for the synced sobjects are seen by the language server as another Apex class. If a user has a small project and syncs 500ish sobjects the project now becomes a larger project to process by the Apex language server and the user might start running into performance problems (depending on their hardware). 

@maaaaarco let me know what you think

Reply:
@lcampos I really appreciate the time for sharing the rationale behind it. I do understand the performance concerns you have and I do agree with them. When working with a newly created project or a small ones, loading those objects is most likely an overkill... On the other hand, I think that forcing these decisions on your users, without giving them the option to opt out, is a bad design pattern. And _I believe_ this to be true for whatever kind of application we build as developers. In my 6 years career as a Salesforce Developer working in consulting business I've seen many projects relying on those objects (especially the custom _Share_ objects).
Long story short, I agree with keeping the filter for most use cases but I do want the option to load all of them... I'm totally happy with the solution you suggested (disable the filter only when the _All SObjects_ option is selected) and I'm keen to update this PR if you would let it go through... We could even add a fourth option one named _Standard and Customs_ that would behave as the _All SObjects_ is doing now (keeping the filter active).
Let me know what do you think ;)

Reply:
@maaaaarco making that small tweak to keep the change exclusive to All SObjects would be enough for us to merge this PR. Adding another option to the menu might not be needed at the moment, we could discuss adding that as a follow up PR but I think it would be almost the same as the existing All SObjects option.

I do agree with you on giving users a way to opt out of certain tooling behavior. Specific to this command, it is one of the least used commands so we are still iterating over it based on the limited feedback and metrics we have. As people use it more we'll keep making changes to it so we get to a point where we have sensible defaults and settings to customize it if needed.

Reply:
@lcampos works for me, I'll have a look at this later next year after 4th of Jan. Again thanks for the background info 👍

Reply:
@maaaaarco sounds good, thanks for contributing to the project. Happy holidays and talk to you next year.